improvement(reversed queries): Added select-order-by to scylla-bench

yarongilor opened the pull request:
The new s-b parameter of select-order-by is added to v0.1.8
            So the new reversed-queries implementation of 5.0 can be tested this way.

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,41 @@
+
+bench_run: true
+
+prepare_write_cmd:  ["scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=8 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=8 -clustering-row-count=50000 -partition-offset=8 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=8 -clustering-row-count=50000 -partition-offset=16 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=8 -clustering-row-count=50000 -partition-offset=24 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=7 -clustering-row-count=50000 -partition-offset=32 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=7 -clustering-row-count=50000 -partition-offset=39 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=7 -clustering-row-count=50000 -partition-offset=46 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=7 -clustering-row-count=50000 -partition-offset=53 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150"
```
Add much more partitions.
instead of 7 or 8 each, make it 100 each.
If the prepare finish less than 2 hours, increase it to even more.

yarongilor replied:
fixed

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,41 @@
+
+nemesis_class_name: 'SisyphusMonkey'
+nemesis_seed: '001'
+instance_type_loader: 'm5n.2xlarge'
```
looks like it it didn't help to overcome the issue in the loader, so change it back to c5n.

yarongilor replied:
fixed

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,41 @@
+
+instance_type_db: 'i4i.4xlarge'
+
+nemesis_class_name: 'SisyphusMonkey'
```
First run it without any nemesis, to observe how it runs and eliminate any interference.

yarongilor replied:
will create a separate debug branch for that.

yarongilor replied:
Ran in: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/11/
and failed. (error not found in logs, probably loader crashed again and got disconnected)

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,41 @@
+primary_key_column: "pk"
+
+
+run_full_partition_scan: '{"ks_cf": "scylla_bench.test", "interval": 3600, "pk_name":"pk", "rows_count": 50000, "validate_data": "true"}' # 'ks.cf, interval(sec), partition-key name, number-of-rows-per-partition, validate reversed query output, include data-column or only validate pk + ck'
```
Comment this out for the first run.
I would like to see only scylla-bench workload in the first run.

yarongilor replied:
Ran in: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/11/
and failed. (error not found in logs, probably loader crashed again and got disconnected)

Argus: https://argus.scylladb.com/test/1b4175fe-4ef3-494c-959a-127a2f105356/runs?additionalRuns[]=0504defa-786c-4368-b954-f1c0eb4a9b9c

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,41 @@
+n_loaders: 2
+n_monitor_nodes: 1
+
+instance_type_db: 'i4i.4xlarge'
```
When the stress finish the 2 days, check what is the disk usage (in monitor) for each node.
We may be able to make it smaller.

yarongilor replied:
will be retested with latest s-b version (not private build).

yarongilor replied:
> When the stress finish the 2 days, check what is the disk usage (in monitor) for each node. We may be able to make it smaller.
@roydahan ,
The disk usage is 5% after one day:
![image](https://user-images.githubusercontent.com/9427929/207285285-45356d20-26e6-48ed-a621-5bfd854dcb0b.png)

The CPU load is around 50 - 70%:

![image](https://user-images.githubusercontent.com/9427929/207284830-65a22b5c-cb89-4d82-aab0-d42aaf207e08.png)

fgelcer replied:
disk utilization is still around 6%

roydahan replied:
Now the question is if we need to retest with 2xlarge or leave it as is.

roydahan commented on test-cases/longevity/scylla-bench-stress-order-by-desc.yaml:
```diff
@@ -0,0 +1,3 @@
+stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=desc -replication-factor=3 -partition-count=60 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400"
```
Also change partition count.

yarongilor replied:
fixed

vponomaryov replied:
Where is it fixed?

yarongilor replied:
fixed in both files now.

yarongilor commented:
Started a draft test run in: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/8/

vponomaryov commented on jenkins-pipelines/longevity-large-partition-2days-reversed-queries.jenkinsfile:
```diff
@@ -0,0 +1,12 @@
+    backend: 'aws',
+    region: 'eu-west-1',
+    test_name: 'longevity_large_partition_test.LargePartitionLongevityTest.test_large_partition_longevity',
+    test_config: '''["test-cases/longevity/longevity-large-partition-2days.yaml", "test-cases/longevity/scylla-bench-stress-order-by-desc.yaml"]'''
```
Second files with redefinitions we keep in the "configurations" directory.
"test-cases" must be self-sufficient.

yarongilor replied:
fixed

vponomaryov commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,43 @@
+bench_run: true
+
+prepare_write_cmd:  ["scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=100 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=100 -clustering-row-count=50000 -partition-offset=100 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
```
All the partition offsets are worng.
It must be 101, 201, 301, 401, 501, 601 and 701.
For now, you redefine last one by next loader and do not write last, 800th partition.
It will cause failures.

yarongilor replied:
fixed

fruch commented on configurations/scylla-bench-stress-order-by-desc.yaml:
```diff
@@ -0,0 +1,3 @@
+stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=desc -replication-factor=3 -partition-count=800 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400"
+]
+user_prefix: 'longevity-large-partitions-order-by-desc'
```
would recommend basing it the same:
```suggestion
user_prefix: 'longevity-large-partitions-2d-order-by-desc'
```

regardless this is used for naming the instances, you might want to try keeping short

yarongilor commented on configurations/scylla-bench-stress-order-by-desc.yaml:
```diff
@@ -1,3 +1,3 @@
 stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=desc -replication-factor=3 -partition-count=800 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400"
 ]
-user_prefix: 'longevity-large-partitions-order-by-desc'
+user_prefix: 'longevity-large-partitions-2d-order-by-desc'
```
fixed

yarongilor commented:
@roydahan , @fruch , looks like it's ready for merge.

fruch commented:
>  @roydahan , @fruch , looks like it's ready for merge.

Can you explain a bit why it's ready, and where are the passing runs of those two jobs introduced here...

yarongilor commented:
> > @roydahan , @fruch , looks like it's ready for merge.
> 
> Can you explain a bit why it's ready, and where are the passing runs of those two jobs introduced here...

Last run after comments are addressed (and https://github.com/scylladb/scylla-bench/issues/113 is fixed) is in:
https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/20/
https://argus.scylladb.com/test/1b4175fe-4ef3-494c-959a-127a2f105356/runs?additionalRuns[]=42bbe08b-c76f-4d7a-a019-8aa574535076

One doubt is about the low used disk size that might indicate a smaller instance type is better to use - https://github.com/scylladb/scylla-cluster-tests/pull/5335#discussion_r1046899838

So better wait for this run to end in one day, and decide if a smaller instance type is needed.
Then, following that, we can run the second test job.

yarongilor commented:
> > > > @roydahan , @fruch , looks like it's ready for merge.
> > > 
> > > 
> > > Can you explain a bit why it's ready, and where are the passing runs of those two jobs introduced here...
> > 
> > 
> > Last run after comments are addressed (and [scylladb/scylla-bench#113](https://github.com/scylladb/scylla-bench/issues/113) is fixed) is in: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/20/ [https://argus.scylladb.com/test/1b4175fe-4ef3-494c-959a-127a2f105356/runs?additionalRuns[]=42bbe08b-c76f-4d7a-a019-8aa574535076](https://argus.scylladb.com/test/1b4175fe-4ef3-494c-959a-127a2f105356/runs?additionalRuns%5B%5D=42bbe08b-c76f-4d7a-a019-8aa574535076)
> > One doubt is about the low used disk size that might indicate a smaller instance type is better to use - [#5335 (comment)](https://github.com/scylladb/scylla-cluster-tests/pull/5335#discussion_r1046899838)
> > So better wait for this run to end in one day, and decide if a smaller instance type is needed. Then, following that, we can run the second test job.
> 
> The run completed successfully in 2 days. had about 6% used disk size. (A successive run that did use reversed-queries, failed after 19 hours for: [scylladb/scylla-bench#115](https://github.com/scylladb/scylla-bench/issues/115) )

A full 2 days run completed ok using the fix of https://github.com/scylladb/scylla-bench/pull/117 
(scylla-bench: scylladb/hydra-loaders:scylla-bench-112-fix-channel-leak)
in: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/22/

s-b got some known errors during nemesis like:
```
2023-01-02 22:40:11.572 <2023-01-02 22:40:07.000>: (ScyllaBenchLogEvent Severity.ERROR) period_type=one-time event_id=b37f55cd-1fce-40dc-b347-1f037dd7dc7b during_nemesis=RebuildStreamingErr,DestroyDataThenRebuild: type=ConsistencyError regex=received only line_number=110058 node=Node longevity-large-partitions-2d-order-loader-node-c585588a-1 [3.252.160.230 | 10.4.0.7] (seed: False)
2023/01/02 22:40:07 Operation timed out for scylla_bench.test - received only 1 responses from 2 CL=QUORUM.
```
@fruch , @roydahan , if no further concerns about the above issue or about above disk-utilization amount, this PR can be merged.

fgelcer reviewed (approved):
LGTM

fgelcer commented:
ping @fruch @vponomaryov @roydahan

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,43 @@
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=100 -clustering-row-count=50000 -partition-offset=601 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150",
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=100 -clustering-row-count=50000 -partition-offset=701 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150"
+]
+stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=asc -replication-factor=3 -partition-count=800 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400"
```
Let's add also a write stress during the stress that adds 200 more partitions.

yarongilor replied:
added

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,43 @@
+stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=asc -replication-factor=3 -partition-count=800 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400"
+]
+
+n_db_nodes: 4
```
Let's change to 3 nodes.

yarongilor replied:
changed.

roydahan commented:
On top of my comments, if we got the errors as Yaron mention above, it means that it's not good and need to understand why.

2023-01-02 22:40:11.572 <2023-01-02 22:40:07.000>: (ScyllaBenchLogEvent Severity.ERROR) period_type=one-time event_id=b37f55cd-1fce-40dc-b347-1f037dd7dc7b during_nemesis=RebuildStreamingErr,DestroyDataThenRebuild: type=ConsistencyError regex=received only line_number=110058 node=Node longevity-large-partitions-2d-order-loader-node-c585588a-1 [3.252.160.230 | 10.4.0.7] (seed: False)
2023/01/02 22:40:07 Operation timed out for scylla_bench.test - received only 1 responses from 2 CL=QUORUM.

yarongilor commented on jenkins-pipelines/longevity-large-partition-2days.jenkinsfile:
```diff
@@ -0,0 +1,12 @@
+#!groovy
```
deleted.

yarongilor commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,43 @@
+
+user_prefix: 'longevity-large-partitions-2d'
+
+space_node_threshold: 644245094
```
deleted.

yarongilor commented on configurations/scylla-bench-stress-order-by-desc.yaml:
```diff
@@ -0,0 +1,3 @@
+stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=desc -replication-factor=3 -partition-count=800 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400"
```
removed.

yarongilor commented:
Retesting after addressing comments: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/23/

yarongilor commented:
s-b errors are reproduced oduring several nemesis.
@fruch , @vponomaryov , can you please advise per this comment - https://github.com/scylladb/qa-tasks/issues/415#issuecomment-1404954286

vponomaryov commented:
> Retesting after addressing comments: https://jenkins.scylladb.com/job/scylla-staging/job/Longevity_yaron/job/longevity-large-partition-4days-test-rq2/25/
> 
> Running ok for a few hours. yet s-b gets many timeout errors.

Following doesn't look `running ok` to me:

```
11:37:47  < t:2023-01-25 09:37:45,856 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-01-25 09:37:45.854: (PrometheusAlertManagerEvent Severity.NORMAL) period_type=begin event_id=7ab703f7-aa15-407c-b79a-cff9010f618d: alert_name=HighLatencies node=10.4.0.12 start=2023-01-25T09:15:04.791Z end=2023-01-25T09:41:44.791Z description=10.4.0.12 has average high latency for more than 5 minutes. updated=2023-01-25T09:37:44.821Z state=active fingerprint=37d75254eea9d5d9 labels={'alertname': 'HighLatencies', 'by': 'instance', 'instance': '10.4.0.12', 'level': '1', 'monitor': 'scylla-monitor', 'scheduling_group_name': 'sl:default', 'severity': '1'}
11:45:18  < t:2023-01-25 09:45:04,448 f:nemesis.py      l:1664 c:sdcm.nemesis         p:INFO  > sdcm.nemesis.SisyphusMonkey: >>>>>>>>>>>>>Started random_disrupt_method kill_scylla
11:45:18  < t:2023-01-25 09:45:04,449 f:nemesis.py      l:356  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.SisyphusMonkey: Current Target: Node longevity-large-partitions-2d-order-db-node-f686879c-5 [54.75.54.190 | 10.4.3.59] (seed: False) with running nemesis: DecommissionStreamingErr Node longevity-large-partitions-2d-order-db-node-f686879c-6 [54.154.125.208 | 10.4.2.67] (seed: False)
11:45:18  < t:2023-01-25 09:45:04,451 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-01-25 09:45:04.449: (ClusterHealthValidatorEvent Severity.NORMAL) period_type=begin event_id=597c32da-22f1-4d54-8d8f-d5d1f5427c76
11:45:18  < t:2023-01-25 09:45:08,449 f:cluster.py      l:3363 c:cassandra.cluster    p:INFO  > New Cassandra host <Host: 10.4.3.59:9042 eu-west> discovered
11:45:18  < t:2023-01-25 09:45:08,450 f:cluster.py      l:3363 c:cassandra.cluster    p:INFO  > New Cassandra host <Host: 10.4.0.12:9042 eu-west> discovered
11:45:18  < t:2023-01-25 09:45:13,293 f:cluster.py      l:3363 c:cassandra.cluster    p:INFO  > New Cassandra host <Host: 10.4.0.35:9042 eu-west> discovered
11:45:18  < t:2023-01-25 09:45:13,293 f:cluster.py      l:3363 c:cassandra.cluster    p:INFO  > New Cassandra host <Host: 10.4.0.12:9042 eu-west> discovered
11:45:18  < t:2023-01-25 09:45:17,323 f:cluster.py      l:3363 c:cassandra.cluster    p:INFO  > New Cassandra host <Host: 10.4.0.35:9042 eu-west> discovered
11:45:18  < t:2023-01-25 09:45:17,324 f:cluster.py      l:3363 c:cassandra.cluster    p:INFO  > New Cassandra host <Host: 10.4.3.59:9042 eu-west> discovered
11:45:19  < t:2023-01-25 09:45:19,298 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-01-25 09:45:19.297: (ClusterHealthValidatorEvent Severity.NORMAL) period_type=end event_id=597c32da-22f1-4d54-8d8f-d5d1f5427c76 duration=14s message=Cluster health check finished
11:45:19  < t:2023-01-25 09:45:19,300 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-01-25 09:45:19.297: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=a2f237d8-78d5-40e4-9aae-8fc453904d3e: nemesis_name=KillScylla target_node=Node longevity-large-partitions-2d-order-db-node-f686879c-5 [54.75.54.190 | 10.4.3.59] (seed: False)
11:45:20  < t:2023-01-25 09:45:20,119 f:nemesis.py      l:539  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.SisyphusMonkey: Kill all scylla processes in Node longevity-large-partitions-2d-order-db-node-f686879c-5 [54.75.54.190 | 10.4.3.59] (seed: False)
11:45:20  < t:2023-01-25 09:45:20,530 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-01-25 09:45:20.528: (JMXServiceEvent Severity.NORMAL) period_type=end event_id=0aee935b-9ab6-4118-b840-82698f7aa1b3 duration=4h32m39s node=longevity-large-partitions-2d-order-db-node-f686879c-5
11:45:21  < t:2023-01-25 09:45:21,032 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:INFO  > 2023-01-25 09:45:21.031: (JMXServiceEvent Severity.NORMAL) period_type=begin event_id=91e81452-b873-4341-b643-75aa681e9ba2 node=longevity-large-partitions-2d-order-db-node-f686879c-5
11:45:31  < t:2023-01-25 09:45:30,887 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:ERROR > 2023-01-25 09:45:30.885 <2023-01-25 09:45:30.000>: (ScyllaBenchLogEvent Severity.ERROR) period_type=one-time event_id=b9836868-556b-40e4-abf9-48f4e5e173ee during_nemesis=KillScylla: type=ConsistencyError regex=received only line_number=86873 node=Node longevity-large-partitions-2d-order-loader-node-f686879c-1 [52.212.245.247 | 10.4.0.10] (seed: False)
11:45:31  < t:2023-01-25 09:45:30,887 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:ERROR > 2023/01/25 09:45:30 Operation timed out for scylla_bench.test - received only 1 responses from 2 CL=QUORUM.
```

The `s-b timeout` error started appearing right after the `kill scylla` nemesis killed a Scylla member. 
For 30 minutes before it there were lots of notifications with the `high latencies` statements.
It means that the Scylla cluster was in very bad shape and kill of a 1 more member really caused it to fail to answer the `s-b` requests.
So, I think the `s-b` errors just highlight some other problem here.

fruch commented:
looking at this run @yarongilor is talking about
```
hydra investigate show-monitor f686879c-3a0b-428b-a015-f70b7cc8d914
```

something is total unbalanced there, we have one node with more then 100Gb of ram used
and the 2 other with less then 5Gb
![image](https://user-images.githubusercontent.com/340979/215589011-d4bdff93-66ce-4947-b1a0-1b6792d8b5ef.png)

also seems like one of the 3 nodes doesn't have open CQL connections for the whole duration of the test:
![image](https://user-images.githubusercontent.com/340979/215589681-a6549b82-443f-4414-b8bc-58aac3d273b1.png)

blaming it on s-b is nice, but we need to be a bit more specific

@vponomaryov lets look if we can find a way to enable some of the gocql debug, so we can have a bit more details (i'm guessing it won't be the last time we'll need it)

yarongilor commented:
> looking at this run @yarongilor is talking about
> 
> ```
> hydra investigate show-monitor f686879c-3a0b-428b-a015-f70b7cc8d914
> ```
> 
> something is total unbalanced there, we have one node with more then 100Gb of ram used and the 2 other with less then 5Gb ![image](https://user-images.githubusercontent.com/340979/215589011-d4bdff93-66ce-4947-b1a0-1b6792d8b5ef.png)
> 
> also seems like one of the 3 nodes doesn't have open CQL connections for the whole duration of the test: ![image](https://user-images.githubusercontent.com/340979/215589681-a6549b82-443f-4414-b8bc-58aac3d273b1.png)
> 
> blaming it on s-b is nice, but we need to be a bit more specific
> 
> @vponomaryov lets look if we can find a way to enable some of the gocql debug, so we can have a bit more details (i'm guessing it won't be the last time we'll need it)

Test ran ok with s-b version 16.
@roydahan , @fgelcer , if no other comments, this can be merged.
@fruch , as for the above issue, i think it's better to open a separate task for s-b, because it's not necessarily related to this test.

fgelcer commented:
> > looking at this run @yarongilor is talking about
> > ```
> > hydra investigate show-monitor f686879c-3a0b-428b-a015-f70b7cc8d914
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > something is total unbalanced there, we have one node with more then 100Gb of ram used and the 2 other with less then 5Gb ![image](https://user-images.githubusercontent.com/340979/215589011-d4bdff93-66ce-4947-b1a0-1b6792d8b5ef.png)
> > also seems like one of the 3 nodes doesn't have open CQL connections for the whole duration of the test: ![image](https://user-images.githubusercontent.com/340979/215589681-a6549b82-443f-4414-b8bc-58aac3d273b1.png)
> > blaming it on s-b is nice, but we need to be a bit more specific
> > @vponomaryov lets look if we can find a way to enable some of the gocql debug, so we can have a bit more details (i'm guessing it won't be the last time we'll need it)
> 
> Test ran ok with s-b version 16. [Argus](https://argus.scylladb.com/test/1b4175fe-4ef3-494c-959a-127a2f105356/runs?additionalRuns%5B%5D=c0951f03-f6b7-49bd-b77e-8530d4261137) @roydahan , @fgelcer , if no other comments, this can be merged. @fruch , as for the above issue, i think it's better to open a separate task for s-b, because it's not necessarily related to this test.

@yarongilor , for now, report an issue on SCT for that imbalnce, and we will need to investigate it deeply, as it could be a real Scylla core issue (but could also be an issue with the way the requests are sent, by either the driver, s-b or something else).

fgelcer commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,42 @@
+primary_key_column: "pk"
+
+
+run_full_partition_scan: '{"ks_cf": "scylla_bench.test", "interval": 3600, "pk_name":"pk", "rows_count": 50000, "validate_data": "true"}' # 'ks.cf, interval(sec), partition-key name, number-of-rows-per-partition, validate reversed query output, include data-column or only validate pk + ck'
```
remove unecessary comment

fgelcer reviewed (commented):
small comment + i think the instance is under utilized, but other than it LGTM

fgelcer commented:
@yarongilor , please rebase, and perhaps you will need to change the `run_full_partition_scan` to follow changes made by @temichus

fgelcer reviewed (approved):
LGTM

roydahan commented:
@yarongilor  please check why checks are failing and the last comment from @temichus and ping here so I can merge it.

roydahan commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,42 @@
+instance_type_db: 'i4i.4xlarge'
+
+nemesis_class_name: 'SisyphusMonkey'
+nemesis_seed: '001'
```
```suggestion
nemesis_seed: '156'
```

roydahan replied:
just to use a different seed that we don't use.

yarongilor replied:
fixed.

vponomaryov commented on test-cases/longevity/longevity-large-partition-2days.yaml:
```diff
@@ -0,0 +1,42 @@
+                     "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=100 -clustering-row-count=50000 -partition-offset=701 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150"
+]
+stress_cmd: ["scylla-bench -workload=uniform -mode=read -select-order-by=desc -replication-factor=3 -partition-count=800 -clustering-row-count=50000 -clustering-row-size=uniform:100..5120 -rows-per-request=2000 -timeout=240s -retry-number=10 -concurrency=400 -duration=44h -connection-count 400",
+             "scylla-bench -workload=sequential -mode=write -replication-factor=3 -partition-count=200 -clustering-row-count=50000 -partition-offset=801 -clustering-row-size=uniform:100..5120 -concurrency=250 -rows-per-request=2000 -timeout=180s -connection-count 150"
```
Shouldn't also this one use `select-order-by=desc`?

yarongilor replied:
@vponomaryov , `select-order-by=desc` only applies to read mode.

fgelcer reviewed (approved):
LGTM

vponomaryov reviewed (approved):
LGTM
